### PR TITLE
pkg/lightning: reduce cpu&memory usage when duplicate-resolution is enabled

### LINKED ISSUE #29564: lightning: import is too slow when duplicate-detect is enabled
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
Run lightning with `duplication-resolution = "record"`

### 2. What did you expect to see? (Required)
The import speed should be around 300GB/h.

### 3. What did you see instead (Required)
The actual import speed is  around 100GB/h. 

### 4. What is your TiDB version? (Required)
the newest commit of release-5.3

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?
close #29564

### What is changed and how it works?
Fix several bugs and performance issues.
* Fix the issue that `hasDupe` is always true.
* Fix the issue that `logger.With` cost too much CPU.
* Fix the issue that `ResolveDuplicateRows` causes TiDB OOM.
* Fix the issue that multiple Lightnings call `ResolveDuplicateRows`.
* Remove `switchToNormalMode` after importing is finished.

### Check List

Tests 

- [x] Manual test (add detailed scripts or steps below)
Modify `GetConflictKeys(ctx, tableName, preRowID, 1000)` to `GetConflictKeys(ctx, tableName, preRowID, 1)` to force resolution one row at a time. Then manual importing `br/tests/lightning_duplicate_detection/data` with `duplicate-resolution=remove`. After importing was finished, all duplicates were removed and all table can pass `admin check table xxx`.

### Release note

```release-note
None
```

### CODE REVIEW

**Review by Little-Wallace — APPROVED** [review]
LGTM

File: br/pkg/lightning/restore/table_restore.go [hunk 1]
```diff
@@ -787,6 +780,19 @@ func (tr *TableRestore) postProcess(
...
 				}
 			}
+		} else {
+			switch {
+			case rc.cfg.PostRestore.Checksum == config.OpLevelOff:
+				tr.logger.Info("skip checksum because the checksum option is off")
+			case hasDupe:
+				tr.logger.Info("skip checksum&analyze because duplicates were detected")
+				shouldSkipAnalyze = true
+			case needChecksum:
```
Comment (glorv) [inline br/pkg/lightning/restore/table_restore.go:790]:
```suggestion
			case !needChecksum:
```

**Review by glorv — COMMENTED** [review]
rest LGTM

File: br/pkg/lightning/restore/table_restore.go (hunk 1 above)
Comment (gozssky) [inline br/pkg/lightning/restore/table_restore.go:790]:
```suggestion
			case !needChecksum:
```

**Review by glorv — APPROVED** [review]

**Pull request merged**